components/src/core/components/List/types.ts

```diff
@@ -0,0 +1,126 @@
+interface Visible {
+  visible?: boolean;
+}
+
+interface ListEmitEvent {
+  type: string;
+  identifier: string;
+}
+
+interface Tooltip {
+  tooltip?: string;
+  flow?: string;
+}
+
+interface Style {
+  [key: string]: string | undefined;
+}
+
+interface Icon {
+  iconName?: string;
+  "icon-size"?: string;
```

Comment:
Why do we use "kebab-case" here?

Reply:
Lint forced to use kebab-case for interface names when running lint checks on github

Reply:
updated with camel case and made relevant reference changes on Vue App